app/api/constituency_lookup/[postcode]/[[...address]]/route.ts

```diff
@@ -84,16 +135,54 @@ export async function GET(
     };
     return NextResponse.json(response);
   } else {
-    // TODO: Use DemocracyClub API to lookup postcode and populate the addresses array, so users can select their
-    // specific address, rather than us expecting to know (or find out) their constituency.
     console.log(
       `Multiple constituencies found for postcode ${normalizedPostcode}`,
     );
+
     const response: ConstituencyLookupResponse = {
       postcode: params.postcode,
       addressSlug: params.address?.[0],
       constituencies: constituencies,
     };
+
+    //see if DC api will give us anything useful
+    //TODO might be worth setting a timeout on this?
+    //TODO remove this if statement, (here so we can see DC API failure!)
+    //let dc_data = null;
+    //if (normalizedPostcode != "DE30GU")
+    // dc_data = await fetch_dc_api(normalizedPostcode, params.address?.[0]);
+
+    const dc_data = await fetch_dc_api(normalizedPostcode, params.address?.[0]);
```

Comment:
As mentioned in your TODO above, think it's deffo worth adding a timeout to this, and if DC API fails we fallback to our current "manually choose your constituency" approach. During the locals, the DC API did fall over (because of us...!). Obvs we're hitting it much less frequently now, but other orgs might be using it and get big spikes, which seems like it could feasibly bring the DC API down (or slow it significantly).